### ISSUE TITLE
Singularity support

### ISSUE THREAD

**Issue by abol-karimi** [ApolloAuto/apollo#15331]
I want to evaluate Apollo for our research project on scenario-based testing. I need to run Apollo in an HPC cluster which doesn't allow Docker. It would be great if a Singularity-based setup was provided.

**Comment by daohu527** [comment]
Do you need to modify the code and test it on the fly, or do you just want a release version? 

At present, the workload of removing containers is relatively large, so it may be difficult to achieve in the short term, but it is indeed necessary to release a release version. 

I suggest that if it is a LAN, you can simulate it through the network.

**Comment by abol-karimi** [comment]
I just need a release version, for testing similar to [BehAVExplor](https://github.com/MingfeiCheng/BehAVExplor) and [ABLE](https://dl.acm.org/doi/abs/10.1145/3597926.3598108). The HPC cluster is UNC's [Longleaf](https://help.rc.unc.edu/longleaf-cluster/) and I connect to it remotely via UNC's VPN.

**Comment by daohu527** [comment]
Then another question, does HPC cluster support linux? Currently, Apollo is mainly verified under ubuntu. I am not sure about other linux distributions. Does your environment support linux?

**Comment by abol-karimi** [comment]
Yes, the nodes on the cluster run Red Hat Enterprise Linux. I run my programs on them as an Ubuntu image (using Singularity). I was able to port [Carla's docker image](https://github.com/carla-simulator/carla/blob/master/Util/Docker/Release.Dockerfile) to Singularity and run it on the cluster. For Apollo, I need to also port the [Carla-Apollo-Bridge](https://github.com/guardstrikelab/carla_apollo_bridge) setup to Singularity, since I mainly work with Carla and [Scenic](https://github.com/BerkeleyLearnVerify/Scenic).
